### PR TITLE
[FEATURE] Test Framework

### PR DESCRIPTION
**Pull request by aaronsaray**
#### Issue or feature explanation
Please see issue #43 

#### Proposed solution/change
Please look through the commits to understand the changes.  A few things to note:
* to organize things properly, I moved around the PHPMD configuration
* I locked php versions and used the proper testbench version
* I made an example of a feature and a unit test
* I made a change to determine the driver - based on the driver value, not the name of the configuration.

### CODE REVIEW

**Comment by QuintenJustus** [comment]
Looks awesome, very nice job! Thanks a lot for your help!

File: tests/Feature/IndexTest.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        $response->assertViewIs('Prequel::error');
+        $response->assertSeeText('Error in Prequel');
+        $response->assertSeeText('Prequel has been disabled.');
+    }
+
+    // additional tests should be ran that indicate successful connections
+
+    // I don't recommend testing with app environment is 'production' because that may cause other side effects
+    // instead, in the future, create a method that can be mocked that returns whether we're in production or not.
+}
```
Comment (xuanquynh) [inline tests/Feature/IndexTest.php]: We should leave an empty line each php file.

File: tests/Unit/Classes/Database/SequelAdapterTest.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+declare(strict_types=1);
+
+namespace Protoqol\Prequel\Tests\Unit\Classes\Database;
+
+use Protoqol\Prequel\Classes\Database\SequelAdapter;
+use Protoqol\Prequel\Tests\TestCase;
+
+/**
+ * Class SequelAdapterTest
+ * @package Protoqol\Prequel\Tests\Unit\Classes\Database
```
Comment (xuanquynh) [inline tests/Unit/Classes/Database/SequelAdapterTest.php]: These comments aren't necessary, are they?

Comment (aaronsaray) [reply tests/Unit/Classes/Database/SequelAdapterTest.php]: They're not necessary, but the coding standard I use requests them to be added.  There is no requested coding standard on this project yet - if you add one, I'll make sure future contributions match.

File: composer.json [hunk 3]
```diff
@@ -32,5 +34,17 @@
     "psr-4": {
       "Protoqol\\Prequel\\Tests\\": "tests/"
     }
+  },
+  "scripts": {
+    "phpmd": [
+      "@php vendor/bin/phpmd src html cleancode,codesize,controversial,design,naming,unusedcode > public/phpmd.html"
+    ],
+    "test": [
+      "@php vendor/bin/phpunit"
```
Comment (xuanquynh) [inline composer.json]: You can only use "phpmd src html" and "phpunit". Composer knows the bin-dir to find executable files.

Comment (aaronsaray) [reply composer.json]: Oh cool - I didn't know that - thanks! Updating.

File: tests/Unit/Classes/Database/SequelAdapterTest.php (hunk 2 above)
Comment (QuintenJustus) [reply tests/Unit/Classes/Database/SequelAdapterTest.php]: See CONTRIBUTING.md, PSR-2 is the current standard. The class comment is correct.

Comment (aaronsaray) [reply tests/Unit/Classes/Database/SequelAdapterTest.php]: Whoops - I missed that. I normally just follow whatever is in a phpcs.xml doc - and I didn't think to look at that file. Sorry about that. :)

Comment (QuintenJustus) [reply tests/Unit/Classes/Database/SequelAdapterTest.php]: No problem!

File: tests/Feature/IndexTest.php (hunk 1 above)
Comment (aaronsaray) [reply tests/Feature/IndexTest.php]: I have added blank lines according to PSR-2 - good catch.

**Comment by QuintenJustus** [comment]
Changed target branch to Protoqol:Dev - the where the next release lives.

**Pull request merged**